src/phasorpy/phasor.py

```diff
@@ -3070,11 +3110,157 @@ def _median(
     return numpy.nanmedian(real, **kwargs), numpy.nanmedian(imag, **kwargs)
 
 
-def _median_filter(
-    real: ArrayLike,
-    imag: ArrayLike,
+def _median_filter_2d(
+    real: NDArray[Any],
+    imag: NDArray[Any],
+    axes: Sequence[int],
+    /,
+    *,
     repeat: int = 1,
-    size: int | tuple[int] | None = 3,
+    size: int = 3,
+    num_threads: int | None = None,
+) -> tuple[NDArray[Any], NDArray[Any]]:
+    """Return the phasor coordinates after applying a 2D median filter.
+
+    Parameters
+    ----------
+    real : ndarray
+        Real components of the phasor coordinates.
+    imag : ndarray
+        Imaginary components of the phasor coordinates.
+    axes : sequence of int
+        Axes along which to apply the filter.
+    repeat : int, optional
+        Number of times to apply filter. The default is 1.
+    size : int, optional
+        The size of the median filter kernel. Default is 3.
+    num_threads : int, optional
+        Number of OpenMP threads to use for parallelization.
+        By default, multi-threading is disabled.
+        If zero, up to half of logical CPUs are used.
+        OpenMP may not be available on all platforms.
+
+    Returns
+    -------
+    real : ndarray
+        Filtered real component of phasor coordinates.
+    imag : ndarray
+        Filtered imaginary component of phasor coordinates.
+
+    """
+    real = numpy.asarray(real)
+    if real.dtype == numpy.float32:
+        real = real.copy()
+    else:
+        real = real.astype(float)
+
+    imag = numpy.asarray(imag)
+    if imag.dtype == numpy.float32:
+        imag = imag.copy()
+    else:
+        imag = imag.astype(float)
+
+    if len(axes) != 2:
+        return _median_filter_nd(real, imag, axes, repeat=repeat, size=size)
+
+    num_threads = number_threads(num_threads)
+
+    for _ in range(repeat):
+        for index in numpy.ndindex(
+            *[real.shape[ax] for ax in range(real.ndim) if ax not in axes]
+        ):
+            index_list: list[int | slice] = list(index)
+            for ax in axes:
+                index_list = index_list[:ax] + [slice(None)] + index_list[ax:]
+            full_index = tuple(index_list)
+
+            real_slice = real[full_index]
+            imag_slice = imag[full_index]
```

Comment:
Maybe I missed something before, but it looks like `real_slice` and `imag_slice` are not really guaranteed to be C contiguous. So the Cython type would have to be reverted to `const float_t[:, :] image`. Do the tests cover non-contiguous axes and slices?

Reply:
You are right Christoph. In the most common use I think they are, but not necessarily for any case. Maybe it's a good idea to revert this just in case. The alternative would be to allow only contiguous but that would limit the application. I'm inclined to revert to `const float_t[:, :] image`. What do you think?

Also, I will some test cases for non-contiguous axes/slices.